ocaml/xapi/nm.ml

```diff
@@ -473,7 +473,12 @@ let bring_pif_up ~__context ?(management_interface=false) (pif: API.ref_PIF) =
         if mtu <> rc.API.pIF_MTU then
           Db.PIF.set_MTU ~__context ~self:pif ~value:mtu
       with _ ->
-        warn "could not update MTU field on PIF %s" rc.API.pIF_uuid
+        warn "could not update MTU field on PIF %s" rc.API.pIF_uuid;
+      
+      (* sync igmp_snooping_enabled *)
+      let igmp_snooping = Db.Pool.get_igmp_snooping_enabled ~__context ~self:(Helpers.get_pool ~__context) in
+      if igmp_snooping <> rc.API.pIF_igmp_snooping_enabled then
+        Db.PIF.set_igmp_snooping_enabled ~__context ~self:pif ~value:igmp_snooping
```

Comment:
Hmm... I think that we should also update the VLANs that are on top of this PIF, if this PIF has `VLAN_slave_of <> []`. This because the VLANs are essentially just ports on the "parent" bridge.